File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // two pointer approach
        // one to keep insertion location, the other to find next non-duplicate to insert
        int l = 0;
        int r = 0;
        int size = nums.size();
        
        while (r < size) {
            while (r < size && nums[r] == nums[l]) r++;
            l++;
            if (r < size) nums[l] = nums[r];
        }
        
        return l;
    }
};

File: 26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l,r = 0,0
        
        while r < len(nums):
            while r < len(nums) and nums[r] == nums[l]:
                r+=1
            
            if r < len(nums):
                l+=1
                nums[l] = nums[r]
                
        return l+1